src/samVisualize/graphGeneration/cfnTemplateGraphGenerator.ts

```diff
@@ -0,0 +1,138 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Graph, GraphObject } from './graph'
+import { yamlParse } from 'yaml-cfn'
+import { LinkTypes } from './linkTypes'
+import { isString, isObjectLike } from 'lodash'
+
+/**
+ * @param inputYaml A string representing a YAML template
+ * @returns A JavaScript object corresponding to the given YAML string
+ */
+function yamlStringToObject(inputYaml: string): Record<string, unknown> {
+    let templateAsObject = {}
+    try {
+        templateAsObject = yamlParse(inputYaml)
+    } catch (err) {
+        console.error(err)
```

Comment:
Use `getLogger().error` for recording errors. It would be nice to include some extra detail about the error, e.g. 
```js
getLogger().error(`SAM Visualize: failed to load template "${inputYaml}": %O`, err)
```

These show up in multiple places, one of which can be found using the command `AWS: View Toolkit Logs`

Reply:
For sure. Although, `inputYaml` is the entire template contents as a string, so maybe we want to truncate that in the error message? Something like `SAM Visualize: failed to load template "${inputYaml.substr(0,25)}"... Please ensure the template is valid YAML`

Reply:
Ah I thought it was the path. Yeah some way to identify it would be ideal (truncation works)